### PR TITLE
Feature frontend landingpage login signup

### PR DESCRIPTION
**Pull request by nishthaarya**
Added login page, signup page
Redux base for auth
Landing page 60%

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@material-tailwind/react": "^1.2.3",
+    "@reduxjs/toolkit": "^1.8.3",
+    "@testing-library/jest-dom": "^5.16.4",
+    "@testing-library/react": "^13.3.0",
+    "@testing-library/user-event": "^13.5.0",
+    "axios": "^0.27.2",
+    "email-validator": "^2.0.4",
```
Comment (jesusverma) [inline package.json]: lets use custom code

File: package.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      "not op_mini all"
+    ],
+    "development": [
+      "last 1 chrome version",
+      "last 1 firefox version",
+      "last 1 safari version"
+    ]
+  },
+  "devDependencies": {
+    "autoprefixer": "^10.4.8",
```
Comment (jesusverma) [inline package.json:51]: use prettier , eslint etc in devdep

File: src/App.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import Routing from "./routes/Routing";
+import {BrowserRouter as Router} from "react-router-dom"
+
+function App() {
+  return (
+    <Router>
```
Comment (jesusverma) [inline src/App.js:6]: lets have different file for route

Comment (jesusverma) [inline src/App.js:8]: suspension and eerror boudnary implementation

File: src/components/Landing/Features.jsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                            </div>
+                        </div>
+                        <div className="lg:w-1/2">
+
+                        </div>
+                    </div>
+                    <div className="grid gap-8 row-gap-10 sm:grid-cols-2 lg:grid-cols-3">
+                        <div className="bg-pink max-w-md border-2 border-black rounded p-2">
+                            <div className="flex items-center justify-center w-16 h-16 mb-4 rounded-full bg-pink">
+                                <svg
```
Comment (jesusverma) [inline src/components/Landing/Features.jsx:21]: icon should go in asset folder

File: src/components/Signup.jsx [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+            let config = {
+                method: "post",
+                url: "https://vanity-manual.el.6yr.appspot.com/api/v1/users/signup",
+                headers: {
+                    "Content-Type": "application/json"
+                },
+                data: payload
+            }
+
+            axios(config)
```
Comment (jesusverma) [inline src/components/Signup.jsx]: create a wrapper for calling api's

File: src/components/Signup.jsx [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+                "city": city,
+                "gender": gender,
+                "budget": budget,
+                "password": password,
+                "rating": 0
+            }
+
+            let config = {
+                method: "post",
+                url: "https://vanity-manual.el.6yr.appspot.com/api/v1/users/signup",
```
Comment (jesusverma) [inline src/components/Signup.jsx]: base url should go in config or env

File: src/components/Signup.jsx [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+
+const Signup = () => {
+
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+    const [name, setName] = useState("");
+    const [phone, setPhone] = useState("");
+    const [budget, setBudget] = useState("");
+    const [gender, setGender] = useState("");
+    const [city, setCity] = useState("");
```
Comment (jesusverma) [inline src/components/Signup.jsx]: can we wrap this all in an object.
objec.email
objec.passwrod
    const [user, setUser] = useState({});

File: src/pages/Landing/index.jsx [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import Features from "../../components/Landing/Features";
+import Hero from "../../components/Landing/Hero";
+
+const Landing = () => {
+    return (
```
Comment (jesusverma) [inline src/pages/Landing/index.jsx:7]: Error boundary for root file of a module

**Comment by nishthaarya** [comment]
All the changes reviewed have been made and pushed. Merging to deploy on heroku and proceed with api integrations.
@jesusverma

**Pull request merged**